packages/proxy/lib/http/util/service-worker-manager.ts

```diff
@@ -0,0 +1,155 @@
+import type { BrowserPreRequest } from '../../types'
+
+type ServiceWorkerRegistration = {
+  registrationId: string
+  scopeURL: string
+  activatedServiceWorker?: ServiceWorker
+}
+
+type ServiceWorker = {
+  registrationId: string
+  scriptURL: string
+  initiatorURL?: string
+  controlledURLs: Set<string>
+}
+
+type RegisterServiceWorkerOptions = {
+  registrationId: string
+  scopeURL: string
+}
+
+type UnregisterServiceWorkerOptions = {
+  registrationId: string
+}
+
+type AddActivatedServiceWorkerOptions = {
+  registrationId: string
+  scriptURL: string
+}
+
+type AddInitiatorToServiceWorkerOptions = {
+  scriptURL: string
+  initiatorURL: string
+}
+
+/**
+ * Manages service worker registrations and their controlled URLs.
+ *
+ * The basic lifecycle is as follows:
+ *
+ * 1. A service worker is registered via `registerServiceWorker`.
+ * 2. The service worker is activated via `addActivatedServiceWorker`.
+ *
+ * At some point while 1 and 2 are happening:
+ *
+ * 3. We receive a message from the browser that a service worker has been registered with the `addInitiatorToServiceWorker` method.
```

Comment:
registered or initiated? 
```suggestion
 * 3. We receive a message from the browser that a service worker has been initiated with the `addInitiatorToServiceWorker` method.
```

Reply:
Resolved here: [`04617eb` (#28517)](https://github.com/cypress-io/cypress/pull/28517/commits/04617eb1b3565c380ac89d12f575f2e731cba4f6)